File: src/Python/Gencode/syminit.py
from sympy import *
from numpy import *

def syminit(app):

    nd = app['nd'];
    nc = app['nc'];
    ncx = app['ncx'];
    ncu = app['ncu'];
    ncq = app['ncq'];
    ncp = app['ncp'];
    nce = app['nce'];
    nco = app['nco'];
    ncw = app['ncw'];
    ntau = len(app['tau']);
    nuinf = len(app['uinf']);
    nparam = len(app['physicsparam']);

    time = array(symbols("time"));
    xdg = array(symbols("xdg1:" + str(ncx+1)));
    udg = array(symbols("udg1:" + str(nc+1)));
    udg1 = array(symbols("udgp1:" + str(nc+1)));
    udg2 = array(symbols("udgm1:" + str(nc+1)));
    wdg = array(symbols("wdg1:" + str(ncw+1)));
    wdg1 = array(symbols("wdgp1:" + str(ncw+1)));
    wdg2 = array(symbols("wdgm1:" + str(ncw+1)));
    odg = array(symbols("odg1:" + str(nco+1)));
    odg1 = array(symbols("odgp1:" + str(nco+1)));
    odg2 = array(symbols("odgm1:" + str(nco+1)));
    uhg = array(symbols("uhg1:" + str(ncu+1)));
    nlg = array(symbols("nlg1:" + str(nd+1)));
    tau = array(symbols("tau1:" + str(ntau+1)));
    uinf = array(symbols("uinf1:" + str(nuinf+1)));
    param = array(symbols("param1:" + str(nparam+1)));

    return xdg, udg, udg1, udg2, wdg, wdg1, wdg2, odg, odg1, odg2, uhg, nlg, tau, uinf, param, time

File: src/Julia/Gencode/syminit.jl
function syminit(app)

nd = app.nd;
nc = app.nc;
ncx = app.ncx;
ncu = app.ncu;
ncq = app.ncq;
ncp = app.ncp;
nce = app.nce;
nco = app.nco;
ncw = app.ncw;
ntau = length(app.tau);
nuinf = length(app.uinf);
nparam = length(app.physicsparam);

time = SymPy.symbols("time");

xdg = [SymPy.symbols("xdg$i") for i=1:ncx];

udg = [SymPy.symbols("udg$i") for i=1:nc];
udg1 = [SymPy.symbols("udg1$i") for i=1:nc];
udg2 = [SymPy.symbols("udg2$i") for i=1:nc];

wdg = [SymPy.symbols("wdg$i") for i=1:ncw];
wdg1 = [SymPy.symbols("wdg1$i") for i=1:ncw];
wdg2 = [SymPy.symbols("wdg2$i") for i=1:ncw];

odg = [SymPy.symbols("odg$i") for i=1:nco];
odg1 = [SymPy.symbols("odg1$i") for i=1:nco];
odg2 = [SymPy.symbols("odg2$i") for i=1:nco];

uhg = [SymPy.symbols("uhg$i") for i=1:ncu];
nlg = [SymPy.symbols("nlg$i") for i=1:nd];

tau = [SymPy.symbols("tau$i") for i=1:ntau];
uinf = [SymPy.symbols("uinf$i") for i=1:nuinf];
param = [SymPy.symbols("param$i") for i=1:nparam];

return xdg, udg, udg1, udg2, wdg, wdg1, wdg2, odg, odg1, odg2, uhg, nlg, tau, uinf, param, time

end
